pkg/daprrp/processors/statestores/processor_test.go

```diff
@@ -199,6 +200,111 @@ func Test_Process(t *testing.T) {
 		require.Equal(t, []unstructured.Unstructured{*generated}, components.Items)
 	})
 
+	t.Run("success - manual - with secret store", func(t *testing.T) {
+		processor := Processor{
+			Client: k8sutil.NewFakeKubeClient(scheme.Scheme, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "test-namespace"}}),
+		}
+
+		resource := &datamodel.DaprStateStore{
+			BaseResource: v1.BaseResource{
+				TrackedResource: v1.TrackedResource{
+					Name: "some-other-name",
+				},
+			},
+			Properties: datamodel.DaprStateStoreProperties{
+				BasicResourceProperties: rpv1.BasicResourceProperties{
+					Application: applicationID,
+				},
+				BasicDaprResourceProperties: rpv1.BasicDaprResourceProperties{
+					ComponentName: componentName,
+				},
+				ResourceProvisioning: portableresources.ResourceProvisioningManual,
+				Metadata: map[string]any{
+					"config": "extrasecure",
+					"connectionString": map[string]any{
+						"secretKeyRef": map[string]any{
+							"name": "secretStoreKey",
+							"key":  "secretStoreValue",
+						},
+					}},
+				Resources:                []*portableresources.ResourceReference{{ID: externalResourceID1}},
+				Type:                     "state.redis",
+				Version:                  "v1",
+				SecretStoreComponentName: secretStoreComponentName,
+			},
+		}
+
+		options := processors.Options{
+			RuntimeConfiguration: recipes.RuntimeConfiguration{
+				Kubernetes: &recipes.KubernetesRuntime{
+					Namespace: "test-namespace",
+				},
+			},
+		}
+
+		err := processor.Process(context.Background(), resource, options)
+		require.NoError(t, err)
+
+		require.Equal(t, componentName, resource.Properties.ComponentName)
+
+		expectedValues := map[string]any{
+			"componentName": componentName,
+		}
+		expectedSecrets := map[string]rpv1.SecretValueReference{}
+
+		expectedOutputResources, err := processors.GetOutputResourcesFromResourcesField(resource.Properties.Resources)
+
+		generated := &unstructured.Unstructured{
+			Object: map[string]any{
+				"apiVersion": dapr.DaprAPIVersion,
+				"kind":       dapr.DaprKind,
+				"metadata": map[string]any{
+					"namespace":       "test-namespace",
+					"name":            "test-component",
+					"labels":          kubernetes.MakeDescriptiveDaprLabels("test-app", "some-other-name", dapr_ctrl.DaprStateStoresResourceType),
+					"resourceVersion": "1",
+				},
+				"spec": map[string]any{
+					"type":    "state.redis",
+					"version": "v1",
+					"metadata": []any{
+						map[string]any{
+							"name":  "config",
+							"value": "extrasecure",
+						},
+						map[string]any{
+							"name": "connectionString",
+							"secretKeyRef": map[string]any{
+								"name": "secretStoreKey",
+								"key":  "secretStoreValue",
+							},
+						},
+					},
+				},
+				"auth": map[string]any{
+					"secretStore": secretStoreComponentName,
+				},
```

Comment:
Is this `auth` generated by **Dapr** since we create a **Dapr** object with a property called **secretStoreComponentName**?

Reply:
In this case, the auth.secretstore property is added [here](https://github.com/radius-project/radius/pull/7823/files/4b23f1c2bb1e783d06e3d11a34c2296dbb512e85#diff-6ae8441cbc7aeb3a6c3a1d2dc2e278e964ddf0f99649c07832c7bf133f436e7dR100-R107) if the Applications.Dapr/* contains the `secretStoreComponentName` property.
The auth.secretstore property comes from the [Dapr component spec](https://github.com/dapr/dapr/blob/master/charts/dapr/crds/components.yaml#L26)